lib/publiccloud/azure.pm

```diff
@@ -215,12 +233,20 @@ sub upload_img {
 
         # Print image definitions as a help to debug possible conflicting definitions
         my $images = script_output("az sig image-definition list -g '$resource_group' -r '$gallery'");
-        record_info("img-def", "Existing image definitions:\n$images");
-
-        # Note: Repetitive calls are do not fail
-        assert_script_run("az sig image-definition create --resource-group '$resource_group' --gallery-name '$gallery' " .
-              "--gallery-image-definition '$definition' --os-type Linux --publisher '$publisher' --offer '$offer' --sku '$sku' " .
-              "--architecture '$arch' --hyper-v-generation '$hyperv' --os-state 'Generalized'", timeout => 300);
+        record_info("img-defs", "Existing image definitions:\n$images");
+
+        # Create new definition only, if there is no previous definition present
+        my $definition = $self->get_image_definition($resource_group, $gallery, $publisher, $offer, $sku);
+        if (defined $definition) {
+            record_info("use img-def", "Using found image definitions:\n$definition");
+        } else {
+            $definition = get_var("PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION", uc(get_required_var('DISTRI') . '-' . get_required_var('FLAVOR') . '-' . get_required_var('VERSION') . '-' . $sku));
```

Comment:
why we not calling here `gen_image_definition` ?

Reply:
good catch!